pkg/query/api/v1.go

```diff
@@ -101,8 +103,10 @@ type API struct {
 	queryableCreate query.QueryableCreator
 	queryEngine     *promql.Engine
 
-	instantQueryDuration   prometheus.Histogram
-	rangeQueryDuration     prometheus.Histogram
+	instantQueryDuration prometheus.Histogram
+	rangeQueryDuration   prometheus.Histogram
+	// Total number of query failures grouped by type of error.
+	queryFailures          *prometheus.CounterVec
```

Comment:
This is a bit of duplicate to the gRPC server metrics, so not sure. I guess GRPC will not show us per error type.

Reply:
Also what about having separate to `range` and `instant` to be consistent with duration historgrams above?